source/server.js

```diff
@@ -51,3 +52,28 @@ class ProtocolSocket {
         });
     }
 }
+
+////////////////////////////////////////////////
+// Read messages from a socket synchronously. //
+////////////////////////////////////////////////
+
+class SynchronousSocket {
+    constructor(socketPath, scriptPath=null, timeout=null) {
+        this.socketPath = socketPath;
+        this.scriptPath = scriptPath || __dirname + "/query.js";
+        this.timeout = this.timeout || undefined;
+    }
+    request (data) {
+        var response = childProcess.execFileSync("node", [ this.scriptPath ], {
```

Comment:
I don't believe so. Electron demands a visible screen and has a much more expensive startup procedure. Hence why I chose node.

Reply:
With regards to fork, I have been looking into it. From what I understand, `fork` is not actually `fork` as we know it from C.

Reply:
It is potentially possible to use a child process, then pipe the result to stdout in the child process, and then do a blocking read in the parent process.

Reply:
After a rather thorough investigation, I have concluded, using utility_process/child_process.fork and pipes makes this impossible. One way to do this with forking would be to WRITE the result to a file and then READ it. Unfortunately, that seems even more hacky than execFileSync.

The only true solution would be to write our own extension to Node.js. 

Alternatively, something we could explore in the future is: https://github.com/JacobFischer/netlinkwrapper this would solve our problem for this particular context.